projects/kit/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -0,0 +1,252 @@
+    ChangeDetectionStrategy,
+    Component,
+    inject,
+    Input,
+    Output,
+    ViewChild,
+} from '@angular/core';
+import {takeUntilDestroyed} from '@angular/core/rxjs-interop';
+import {FormsModule} from '@angular/forms';
+import {MaskitoDirective} from '@maskito/angular';
+import type {MaskitoOptions} from '@maskito/core';
+import {maskitoInitialCalibrationPlugin, maskitoTransform} from '@maskito/core';
+import {maskitoGetCountryFromNumber, maskitoPhoneOptionsGenerator} from '@maskito/phone';
+import type {TuiFocusableElementAccessor} from '@taiga-ui/cdk';
+import {
+    AbstractTuiControl,
+    CHAR_PLUS,
+    TuiActiveZoneDirective,
+    tuiAsControl,
+    tuiAsFocusableItemAccessor,
+    tuiIsInputEvent,
+    TuiLetDirective,
+} from '@taiga-ui/cdk';
+import {
+    TUI_TEXTFIELD_SIZE,
+    TuiAppearanceDirective,
+    TuiDataList,
+    TuiDropdownModule,
+    tuiDropdownOptionsProvider,
+    TuiFlagPipe,
+    TuiGroupDirective,
+    TuiHint,
+    TuiPrimitiveTextfieldComponent,
+    TuiPrimitiveTextfieldModule,
+    TuiTextfieldComponent,
+    TuiTextfieldControllerModule,
+    TuiTextfieldModule,
+    tuiTextfieldOptionsProvider,
+} from '@taiga-ui/core';
+import type {TuiCountryIsoCode} from '@taiga-ui/i18n';
+import {TuiChevronDirective} from '@taiga-ui/kit/directives';
+import {TUI_COUNTRIES} from '@taiga-ui/kit/tokens';
+import {PolymorpheusModule} from '@tinkoff/ng-polymorpheus';
+import {validatePhoneNumberLength} from 'libphonenumber-js';
+import type {MetadataJson} from 'libphonenumber-js/core';
+import {getCountryCallingCode} from 'libphonenumber-js/core';
+import {BehaviorSubject, combineLatest, distinctUntilChanged, map, skip} from 'rxjs';
+
+import {TuiGetCountryCallingCodePipe} from './get-country-calling-code.pipe';
+import {TUI_INPUT_PHONE_INTERNATIONAL_OPTIONS} from './input-phone-international.options';
+
+const NOT_FORM_CONTROL_SYMBOLS = /[^+\d]/g;
+
+@Component({
+    standalone: true,
+    selector: 'tui-input-phone-international',
+    imports: [
+        CommonModule,
+        FormsModule,
+        PolymorpheusModule,
+        TuiTextfieldModule,
+        TuiGroupDirective,
+        TuiPrimitiveTextfieldModule,
+        TuiTextfieldControllerModule,
+        TuiHint,
+        TuiDropdownModule,
+        TuiDataList,
+        TuiChevronDirective,
+        TuiAppearanceDirective,
+        TuiActiveZoneDirective,
+        TuiFlagPipe,
+        TuiLetDirective,
+        TuiGetCountryCallingCodePipe,
+        MaskitoDirective,
+    ],
+    templateUrl: './input-phone-international.template.html',
+    styleUrls: ['./input-phone-international.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        tuiAsFocusableItemAccessor(TuiInputPhoneInternationalComponent),
+        tuiAsControl(TuiInputPhoneInternationalComponent),
+        tuiTextfieldOptionsProvider({cleaner: false}),
+    ],
+    host: {
+        '[attr.data-size]': 'textfieldSize.size',
+    },
+    viewProviders: [
```

Comment:
Do we need to isolate it? Sounds like we can just move it to providers?

Reply:
Otherwise, any `tuiDropdown*`-controller will override `limitWidth` to `auto`.

For example:

```html
<tui-input-phone-international
    tuiDropdownDirection="top"
/>
```